[vscode] 2/n Create Empty commands

Ankush-lastmile opened the pull request:
[vscode] 2/n Create Empty commands

This diff adds "create new" aiconfig commands to the vscode extension. Instead of forcing the user to deal with a haiku, they now have the option to start from scratch.

## Testplan

https://github.com/lastmile-ai/aiconfig/assets/141073967/4b314eb0-95cf-4e94-89c5-511ba7580a9f

---
[//]: # (BEGIN SAPLING FOOTER)
Stack created with [Sapling](https://sapling-scm.com). Best reviewed with [ReviewStack](https://reviewstack.dev/lastmile-ai/aiconfig/pull/1448).
* __->__ #1448
* #1447

rholinshead commented on vscode-extension/src/extension.ts:
```diff
@@ -289,6 +298,25 @@ async function createNewAIConfig(
+    if (empty) {
+      return "empty.aiconfig.json";
+    } else {
+      return "untitled.aiconfig.json";
```
nit, should we rename from `untitled` to `example`?

rholinshead commented on vscode-extension/package.json:
```diff
@@ -65,6 +65,16 @@
       },
+      {
+        "command": "vscode-aiconfig.createEmptyAIConfigJSON",
+        "title": "Create New Empty AIConfig (json)",
```
nit, from previous PR - can this just be 'New' without 'Empty'?

rholinshead replied:
Same comment on ordering YAML first here as well

rholinshead commented on vscode-extension/src/extension.ts:
```diff
@@ -241,22 +257,15 @@ export function deactivate() {
 async function createNewAIConfig(
   context: vscode.ExtensionContext,
-  mode: "json" | "yaml" = "json"
+  mode: "json" | "yaml" = "json",
```
Should we default to yaml?

rholinshead commented on vscode-extension/src/extension.ts:
```diff
@@ -289,6 +298,25 @@ async function createNewAIConfig(
 }
 
+function relativeStaticAIConfigFilePath(
+  mode: "json" | "yaml" = "json",
```
same here r.e. yaml as the default

rholinshead commented on vscode-extension/src/extension.ts:
```diff
@@ -289,6 +298,25 @@ async function createNewAIConfig(
+  mode: "json" | "yaml" = "json",
+  empty: boolean = true
+): string {
+  if (mode === "json") {
```
```
const templateConfigName = empty ? "empty" : "example";
return `${templateConfigName}.aiconfig.${mode}`
```
Can be cleaned up quite a bit

rholinshead replied:
Ah, ok, makes sense

rholinshead reviewed (approved):
Accepting to unblock, just minor comments

rholinshead commented on vscode-extension/package.json:
```diff
@@ -202,13 +212,13 @@
             "id": "createAIConfig",
             "title": "Create your first AIConfig",
-            "description": "Turn your VS Code into a generative AI studio\n**Note:** Please complete the ``Initialize Extension`` step before opening an AIConfig file!\n[Create .aiconfig.yaml](command:vscode-aiconfig.createAIConfigYAML)\nIf you prefer JSON:\n[Create .aiconfig.json](command:vscode-aiconfig.createAIConfigJSON)",
+            "description": "Turn your VS Code into a generative AI studio\n**Note:** Please complete the ``Initialize Extension`` step before opening an AIConfig file!\n[Create .aiconfig.yaml](command:vscode-aiconfig.createExampleAIConfigJSON)\nIf you prefer JSON:\n[Create .aiconfig.json](command:vscode-aiconfig.createAIConfigJSON)",
```
```suggestion
            "description": "Turn your VS Code into a generative AI studio\n**Note:** Please complete the ``Initialize Extension`` step before opening an AIConfig file!\n[Create .aiconfig.yaml](command:vscode-aiconfig.createExampleAIConfigYAML)\nIf you prefer JSON:\n[Create .aiconfig.json](command:vscode-aiconfig.createExamleAIConfigJSON)",
```

rholinshead replied:
Can we surface the empty commands here as well?

Ankush-lastmile replied:
just confirming, you want all 4 commands in the walkthrough? I initially thought that would be too many

rholinshead replied:
Maybe just fix these ones and then we can add the new ones later if we get feedback about it. They should be discoverable in the command palette when doing the example commands here

rholinshead reviewed (changes requested):
Ah just noticed the walkthrough so requesting changes so it's seen

Ankush-lastmile commented:
Since we're going in the direction of yaml as a default:
- - Modified Defaults to be yaml
- - swapped ordering of commands to have yaml first

- renamed "empty" command, now just `Create New AIConfig (..)`

> Ah just noticed the walkthrough so requesting changes so it's seen
Addressed and fixed  in 1/n: https://github.com/lastmile-ai/aiconfig/pull/1447#issuecomment-1999951840. Thanks for catching.

rholinshead commented on vscode-extension/package.json:
```diff
@@ -246,6 +256,12 @@
           "command": "vscode-aiconfig.createExampleAIConfigYAML"
+        },
+        {
+          "command": "vscode-aiconfig.createEmptyAIConfigYAML"
```
One of these should be JSON.

Also, if order of these matters we should do Empty before Example and YAML before JSON

Ankush-lastmile replied:
ordering matters above in the section of
```
"contributes": {
    "commands": [
```

With that in mind I don't know what this section does. In the testplan of this diff the 4 create commands work as intended.

Ankush-lastmile replied:
but I'l add it in

Ankush-lastmile replied:
synced offline with @rossdanlm , this is for the new file flow within vscode

rholinshead reviewed (approved):
Accepting to unblock but one of the commands needs fixed to JSON and order should be swapped if it reflects order in the UI
